canonical/assets.ubuntu.com
# Fix Switch not found exception with a proper not found response

## Done

- Upgrade dependencies
- Fix Switch not found exception with a proper not found response: [Sentry issue](https://sentry.is.canonical.com/canonical/assetsubuntucom/issues/32534)
- Add additional unit test

## QA

- Check out this feature branch
- Run dependencies: 
```
docker compose up -d
```
- Run the site using the command `dotrun`
- View the site locally in your web browser at: http://0.0.0.0:8017/not-found.png
- Make sure that the response is:
```json
{
  "code": 404,
  "message": "No asset found for 'not-found.png'"
}
```


### Context

This PR gets rid of this Sentry alert:
<img width="1194" alt="image" src="https://github.com/user-attachments/assets/87cc2f45-8c61-4274-88ae-7f8b61ba5e77">
